imap_processing/tests/mag/test_mag_validation.py

```diff
@@ -0,0 +1,151 @@
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from imap_processing.mag.l1a.mag_l1a import mag_l1a
+from imap_processing.mag.l1a.mag_l1a_data import TimeTuple
+from imap_processing.mag.l1b.mag_l1b import mag_l1b
+
+
+@pytest.mark.parametrize(
+    ("test_number"), ["001", "002", "003", "004", "005", "006", "007", "008"]
+)
+def test_mag_l1a_validation(test_number):
+    source_directory = Path(__file__).parent / "validation" / "L1a" / f"T{test_number}"
+
+    input_file = source_directory / f"mag-l0-l1a-t{test_number}-in.bin"
+    expected_output_file = source_directory / f"mag-l0-l1a-t{test_number}-out.csv"
+
+    mag_l1a_out = mag_l1a(input_file, "v000")
+    expected_output = pd.read_csv(expected_output_file)
+
+    raw = mag_l1a_out[0]
+    primary = mag_l1a_out[1]
+    secondary = mag_l1a_out[2]
+
+    assert len(mag_l1a_out) == 3
+    print(primary["vectors"].data)
+    print(expected_output)
```

Comment:
More prints here - are we okay with print statements in the tests? I usually assume it should be logger statements, but I don't really know if one or another is right.